### PR TITLE
Make agent_groups distributed

### LINKED ISSUE #11789: Modify `agent_groups.py` to distribute the changes performed
## Description

Currently, the script [`agent_groups.py`](https://github.com/wazuh/wazuh/blob/master/framework/scripts/agent_groups.py) changes the information but does not distribute it to the master if executed in the worker. This could lead to a problem of the master node pushing some outdated information to the worker, and removing the newest one.

### PR DESCRIPTION
**Pull request by yanazaeva**
|Related issue|
|---|
|#11789|

## Description

This closes #11789. In this pull request, we made the `agent_groups` binary distributed, meaning that it could be executed from not only the master but the workers too. The master will be in charge of creating, assigning and removing all the groups, and then sending this information to the workers.

### CODE REVIEW

File: framework/scripts/agent_groups.py [hunk 1]
```diff
@@ -138,36 +224,93 @@ def remove_group(group_id, quiet=False):
...
+        Upgrade function
+    command : dict
+        Arguments for the specified function
+    local_master : bool
+        True for get the upgrade results, False for send upgrade command
+
+    Returns
+    -------
+    Distributed API request result
+    """
```
Comment (Selutario) [inline framework/scripts/agent_groups.py:302]:
```suggestion
    """Send the command to the specified function.

    If local_master is True, the request type must be local_master (upgrade_result).

    Parameters
    ----------
    function : Callable
        Upgrade function.
    command : dict
        Arguments for the specified function.
    local_master : bool
        True for get the upgrade results, False for send upgrade command.

    Returns
    -------
    result : AffectedItemsWazuhResult
        Distributed API request result.
    """
```

**Review by Selutario — CHANGES_REQUESTED** [review]

**Review by Selutario — CHANGES_REQUESTED** [review]
Rebase with the epic branch when possible.

File: framework/scripts/agent_upgrade.py [hunk 2]
```diff
@@ -225,4 +226,4 @@ def main():
     except Exception as e:
         print(f"Internal error: {str(e)}")
         if args.debug:
-            raise
+            raise
```
Comment (Selutario) [inline framework/scripts/agent_upgrade.py:229]: Add a blank line at the end
```suggestion
            raise

```

File: framework/scripts/agent_groups.py [hunk 3]
```diff
@@ -36,138 +45,258 @@ def signal_handler(n_signal, frame):
...
 
 def show_group(agent_id):
-    agent_info = agent.get_agents(agent_list=[agent_id]).to_dict()
-    if agent_info['total_affected_items'] == 0:
-        print(list(agent_info['failed_items'].keys())[0])
+    """Show the groups an agent belong to.
+
+    Parameters
+    ----------
+    agent_id: str
```
Comment (Selutario) [inline framework/scripts/agent_groups.py:65]: Remember to leave a space after the variable names in the docstrings
```suggestion
    agent_id : str
```

**Review by Selutario — CHANGES_REQUESTED** [review]

**Review by Selutario — APPROVED** [review]

**Review by davidjiglesias — CHANGES_REQUESTED** [review]
Fix conflicts

File: framework/scripts/agent_groups.py [hunk 4]
```diff
@@ -36,138 +45,258 @@ def signal_handler(n_signal, frame):
...
-        msg = agent.create_group(group_id).dikt['message']
+        result = send_command(function=agent.create_group, command={'group_id': group_id})
+        msg = result.dikt['message']
     else:
         msg = "Cancelled."
 
     print(msg)
 
 
+def send_command(function, command, local_master=True):
```
Comment (davidjiglesias) [inline framework/scripts/agent_groups.py:270]: Function is duplicated with that in agent_upgrade.py

Comment (davidjiglesias) [reply framework/scripts/agent_groups.py:270]: We should add it to utils.py or dapi.py

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**